datafusion/src/physical_optimizer/hash_build_probe_order.rs

```diff
@@ -0,0 +1,291 @@
+// Licensed to the Apache Software Foundation (ASF) under one
```

Comment:
this file also has substantial changes from eef5e2d10e60652aa498925fae3f8c688f5b99fe `src/optimizer/hash_build_probe_order.rs  `

Reply:
I also went through the logic in here and it looks good to me